src/metabase/driver/redshift.clj

```diff
@@ -0,0 +1,97 @@
+                       JOIN pg_attribute source_column ON c.conrelid     = source_column.attrelid
+                       JOIN pg_class dest_table        ON c.confrelid    = dest_table.oid
+                       JOIN pg_attribute dest_column   ON c.confrelid    = dest_column.attrelid
+                     WHERE c.contype            = 'f'::char
+                       AND source_table.relname = ?
+                       AND n.nspname            = ?
+                       AND source_column.attnum = ANY(c.conkey)
+                       AND dest_column.attnum   = ANY(c.confkey)"
+                    (:name table)
+                    (:schema table)])))
+
+(defrecord RedshiftDriver []
+  clojure.lang.Named
+  (getName [_] "Amazon Redshift"))
+
+(extend RedshiftDriver
+  driver/IDriver
+  (merge (sql/IDriverSQLDefaultsMixin)
+         {:date-interval  date-interval
+          :details-fields (constantly [{:name         "host"
+                                        :display-name "Host"
+                                        :placeholder  "my-cluster-name.abcd1234.us-east-1.redshift.amazonaws.com"
+                                        :required     true}
+                                       {:name         "port"
+                                        :display-name "Port"
+                                        :type         :integer
+                                        :default      5439}
+                                       {:name         "db"
+                                        :display-name "Database name"
+                                        :placeholder  "toucan_sightings"
+                                        :required     true}
+                                       {:name         "user"
+                                        :display-name "Master username"
+                                        :placeholder  "cam"
+                                        :required     true}
+                                       {:name         "password"
+                                        :display-name "Master user password"
+                                        :type         :password
+                                        :placeholder  "*******"
+                                        :required     true}])
+          :table-fks      table-fks})
+
+  sql/ISQLDriver
+  (merge postgres/PostgresISQLDriverMixin
+         {:connection-details->spec  connection-details->spec
+          :current-datetime-fn       (constantly (k/sqlfn* :GETDATE))
+          :set-timezone-sql          (constantly nil)
+          :unix-timestamp->timestamp unix-timestamp->timestamp}
+         ;; HACK ! When we test against Redshift we use a session-unique schema so we can run simultaneous tests against a single remote host;
+         ;; when running tests tell the sync process to ignore all the other schemas
+         (when config/is-test?
+           {:excluded-schemas (memoize
```

Comment:
I think what we'd need to do is create a different driver class (e.g `RedshiftTestDriver`) that we use in the place of the originals when running tests. It will be a little tricky making sure the original doesn't get registered in place of the test class (and the test class doesn't get registered in place of the original when running from `lein ring server` or the REPL)